Linux SharedEC perf test uses non-SharedEC binaries

### Describe the bug

```
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/msquic.dll
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/msquic.lib
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/msquic.pdb
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/msquic.pgd
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/secnetperf.exe
Skipped processing item artifacts/bin/windows/x64_Release_schannel_SharedEC/secnetperf.pdb
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/msquic.dll
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/msquic.lib
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/msquic.pdb
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/msquic.pgd
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/secnetperf.exe
Skipped processing item artifacts/bin/windows/x64_Release_schannel_Xdp/secnetperf.pdb
Downloading artifacts/bin/linux/x64_Release_openssl/libmsquic.so to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so
Downloading artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2 to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2
Downloading artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2.1.0 to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2.1.0
Downloading artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so
Downloading artifacts/bin/linux/x64_Release_openssl/secnetperf to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/secnetperf
Downloaded artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so.2.1.0 to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so.2.1.0
Downloaded artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.lttng.so
Downloaded artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2.1.0 to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2.1.0
Downloaded artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2 to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so.2
Downloaded artifacts/bin/linux/x64_Release_openssl/libmsquic.so to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/libmsquic.so
Downloaded artifacts/bin/linux/x64_Release_openssl/secnetperf to /home/test/azpagent/_work/2/s/artifacts/bin/linux/x64_Release_openssl/secnetperf
Total Files: 46, Processed: 6, Skipped: 40, Failed: 0, Download time: 4.04 secs, Download size: 16.695MB
```

The actual files downloaded to the perf machine are regular openssl binaries. Am I missing something?

Linux SharedEC openssl and Linux openssl in perf dashboard are basically the same.

Also, I noticed if Linux sharedEC build completes before regular openssl build, Linux SharedEC perf test will just fail because it won't find binaries to use.

```
Downloading artifacts for build: 318458
Downloading items from container resource #/17030581/artifacts
Downloading artifact artifacts from: https://dev.azure.com/ms//_apis/resources/Containers/17030581?itemPath=artifacts&isShallow=true&api-version=4.1-preview.4
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/libmsquic.lttng.so.2.1.0
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/libmsquic.lttng.so
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/libmsquic.so.2.1.0
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/libmsquic.so.2
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/libmsquic.so
Skipped processing item artifacts/bin/linux/x64_Release_openssl_SharedEC/secnetperf
Total Files: 6, Processed: 0, Skipped: 6, Failed: 0, Download time: 1.015 secs, Download size: 0Bytes
Successfully downloaded artifacts to /home/test/azpagent/_work/2/s
```

```
/usr/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/home/test/azpagent/_work/_temp/41e8af9a-377d-47e1-ac5e-5d95649cf667.ps1'
[06/04/2022 11:21:36] Configuring permissions for artifacts...
chmod: cannot access '/home/test/azpagent/_work/2/s/artifacts/bin': No such file or directory
##[error]PowerShell exited with code '1'.
```


### Affected OS

- [X] All
- [ ] Windows Server 2022
- [ ] Windows 11
- [ ] Windows Insider Preview (specify affected build below)
- [ ] Ubuntu
- [ ] Debian
- [ ] Other (specify below)

### Additional OS information

_No response_

### MsQuic version

main

### Steps taken to reproduce bug

1. Send a PR.

### Expected behavior

Linux SharedEC perf test uses Linux sharedEC build binaries.

### Actual outcome

Linux SharedEC test either fails or use non-sharedEC binaries.

### Additional details

_No response_

--- a/.azure/azure-pipelines.perf.yml
+++ b/.azure/azure-pipelines.perf.yml
@@ -434,6 +434,8 @@ stages:
         arch: ${{ parameters.arch }}
         protocol: ${{ parameters.protocol }}
         logProfile: ${{ parameters.logging }}
+        extraArtifactDir: '_SharedEC'
+        extraTestArgs: -ExtraArtifactDir _SharedEC
         ${{ if ne(parameters.testToRun, 'all') }}:
           testToRun: ${{ parameters.testToRun }}
         testTypes: ${{ parameters.testTypes }}

